tools/run_parallel.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# Uncommenting the next line may be useful when debugging this script.
+#set -x
+
+usage () {
+    cat << EOF
+Usage: run_parallel.sh {command} {num_iter} {num_parallel}
+Where:
+  {command} is a string containing the command, including parameters, to run
+  {num_iter} is an positive integer indicating how many iterations to execute
+  {num_parallel} is an (optional) positive integer indicating how many parallel commands should be executed
+     in each iteration. If not provided, the default is half the number of available CPU cores.
+EOF
+}
+
+if [ "$#" -lt  2 ]; then
+    echo "Illegal number of parameters."
+    usage
+    exit 1
+fi
+
+# Calculate the number of CPU cores. This code is Linux specific at the moment.
+NCORES=$(echo "`grep -c ^processor /proc/cpuinfo` / 2" | bc)
```

Comment:
The variable name indicates the number of cores, while the value calculated is actually "half" the number of cores. How about moving `/ 2` down to line 32?

Reply:
Hi @lukech , that's a reasonable point, and I think moving the divide by 2 would make the code clearer. This is an artifact of when I copied the calculation from the original code, and it used NCORES as the number of cores to use, rather than the number present, the goal being to use half the cores.

Reply:
Got it, thanks for sharing the background.

Reply:
@raviprakashgiri29 , would you like me to make this change? I'm happy to do it, unless you'd prefer to.

Reply:
Yeah, sure. Thanks, @jeremythorp for taking this.

Reply:
Hi @raviprakashgiri29 and @lukech , I've moved the divide by 2 to later in the file, and tidied up the code relevant code.